### PR TITLE
ci: add darglint to ci

### PR DESCRIPTION
**Pull request by bwanglzu**
Add darglint to perform incremental docstring check.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,6 +69,24 @@ jobs:
...
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - id: files
+        uses: jitterbit/get-changed-files@v1
+      - name: docstring check with darlint
+        run: |
+          pip install darglint
+          for changed_file in ${{ steps.files.outputs.all }}; do
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:85]: Where is this coming from?

**Review by cristianmtr — CHANGES_REQUESTED** [review]
Is there a way to test it in this PR with a dummy file that we can then remove?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JoanFM) [reply .github/workflows/ci.yml:85]: from the previous step of the github action, see the `github action workflow` documentation

**Comment by cristianmtr** [comment]
To test

[] add malformed python file
[] add random text file to check that it ignores it

**Comment by bwanglzu** [comment]
added a `test.py`, 
image

**Comment by cristianmtr** [comment]
Issue: it seems like it only checks changed files on the last **sha**, not on entire PR level.

**Comment by bwanglzu** [comment]
@cristianmtr now it should works, can you take a look? I'll remove the unused files.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -69,6 +69,27 @@ jobs:
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - id: file_changes
+        uses: jitterbit/get-changed-files@v1
+      - name: docstring check with darlint
+        run: |
+          pip install darglint
+          pip install pydocstyle
+          for changed_file in ${{ steps.file_changes.outputs.all }}; do
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:88]: Why did you remove the echo? Now in the check you don't see which file it's referring to

```
Successfully installed pydocstyle-5.1.1 snowballstemmer-2.1.0
removeme.py:1 in public class `X`:
        D101: Missing docstring in public class
Error: Process completed with exit code 1.
```

This can be difficult since we have the same filenames across folders

**Review by cristianmtr — CHANGES_REQUESTED** [review]
Also, can you try one other thing:

run the commands on all the files at the same time so we get the status on all of them? As it is, it stops at the first error.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -69,6 +69,27 @@ jobs:
...
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - id: file_changes
+        uses: jitterbit/get-changed-files@v1
+      - name: docstring check with darlint
```
Comment (cristianmtr) [inline .github/workflows/ci.yml:84]:
```suggestion
      - name: docstring check with darglint and pydocstyle
```

**Review by cristianmtr — APPROVED** [review]

**Review by Yongxuanzhang — APPROVED** [review]
